src/acl/AtStep.cc

```diff
@@ -42,12 +42,7 @@ ACLAtStepStrategy::match(ACLData<XactionStep> * &data, ACLFilledChecklist *check
         if (!checklist->request)
             return 0; // we have warned about the missing request earlier
 
-        if (!checklist->request->masterXaction) {
-            debugs(28, DBG_IMPORTANT, "ERROR: Squid BUG: at_step GeneratingCONNECT ACL is missing master transaction info. Assuming mismatch.");
-            return 0;
-        }
-
-        return checklist->request->masterXaction->generatingConnect ? 1 : 0;
+        return checklist->request->generatingConnect ? 1 : 0;
```

Comment:
```suggestion
        return checklist->request->generatingConnect;
```

Reply:
Interesting. I thought an inplicit integral [promotion](https://en.cppreference.com/w/cpp/language/implicit_conversion) should take place here, where true becomes 1.

Reply:
> boolean true does not always implicitly convert to `1`.

It does in C++ [AFAICT](https://stackoverflow.com/a/2725063).

However, for consistency sake, we may return 1s and 0s from match() that returns 0s and 1s in other places.

Reply:
@eduard, you are right. I was thinking of the bool representation which can be the ```-1``` literal.